MajorColor.h

```diff
@@ -0,0 +1,16 @@
+#ifndef MAJOR_COLOR_H_
+#define MAJOR_COLOR_H_
+
+typedef enum {
+    MajorColor_WHITE ,
+    MajorColor_RED,
+    MajorColor_BLACK,
+    MajorColor_YELLOW,
+    MajorColor_VIOLET,
+    MajorColor_TotalNumber
```

Comment:
Nice way to represent the number of enums :)
Is there any way to avoid mismatch between this number and the length of the MajorColorNames array?

Reply:
Sudeep,
i had this question
I am not sure if this is what you are asking for 
`
size = (sizeof(MajorColorNames)/sizeof(MajorColorNames[0]));
assert(size == MajorColor_TotalNumber);
/**
 We can do something like this but this needs to be an run time test and
 For this module or code this would be a kind of startup test...
 Do you have any clue on how to do this at compile time ?
*/
`